cmake/templates/OpenCVConfig-CUDA.cmake.in

```diff
@@ -1,4 +1,8 @@
 # Version Compute Capability from which OpenCV has been compiled is remembered
+option(OpenCV_REQUIRE_CUDA_MINOR_EXACT "Require match the exact cuda minor version to link against produced libraries." ON)
+option(OpenCV_ALLOW_CUDA_MINOR_OLDER "Allow newer cuda minor version to link against produced libraries." OFF)
+option(OpenCV_ENABLE_CUDA_FIRST_CLASS_LANGUAGE "cuda language is enabled" "@ENABLE_CUDA_FIRST_CLASS_LANGUAGE@")
```

Comment:
`OpenCV_ENABLE_CUDA_FIRST_CLASS_LANGUAGE` is internal OpenCV build state, but not user managable option. I propose to use just `set` without a chance to modify it outside.

Reply:
The whole point here is to make it configurable in user projects. The way to find cuda should not be determined at opencv build time. It should be free to choose depending on the user projects.